lib/galaxy/util/properties.py

```diff
@@ -85,15 +85,29 @@ def load_app_properties(
         properties.update(kwds)
 
     # update from env
-    override_prefix = "%sOVERRIDE_" % config_prefix
-    for key in os.environ:
-        if key.startswith(override_prefix):
-            config_key = key[len(override_prefix):].lower()
-            properties[config_key] = os.environ[key]
-        elif key.startswith(config_prefix):
-            config_key = key[len(config_prefix):].lower()
-            if config_key not in properties:
-                properties[config_key] = os.environ[key]
+    override_prefix = "override_"
+    yaml_prefix = "yaml_"
+    for key, val in os.environ.items():
```

Comment:
This "chaining" logic may work for the current set of prefix values. However, it assumes the prefix naming structure will never change ("config" prefix followed by "override" suffix followed by 'yaml" suffix): any significant change would require rewriting the whole function. Could you make it more flexible? Something like
```
if prefix is foo: get config key and do stuff; 
elif prefix is bar: get config key and do other stuff
etc...
```
That way the logic does not depend on the naming structure; also it's easy to add/remove a prefix.

Reply:
I went with a string munching algorithm rather than generating all possible prefixes.
The if..elif chain causes duplicated code. I can switch to that but I don't think it is good form.
Changing the composition order is just a matter of reordering the if blocks in the loop with the current implementation.

Reply:
I don't think we can assume that one prefix will be always a suffix of the other, which is why making the tests for each condition independent may be a more flexible approach.

Reply:
I implemented what you asked but I don't think it is cleaner, easy to maintain, or more flexible.
This will get messy fast if another possible prefix is added.

Reply:
I'm sure we can improve it! So, currently, we have this:
`GALAXY_CONFIG_OVERRIDE_`
`GALAXY_CONFIG_`

We may be adding this:
`GALAXY_CONFIG_OVERRIDE_YAML_`
`GALAXY_CONFIG_YAML_`

However, as we don't want to be locked into one being a prefix of the other (e.g. it could become `GALAXY_CONFIG` + `GALAXY_OVERRIDE`), we can think of them in terms of:
`FOO_`
`BAR_`
`FOO_YAML_`
`BAR_YAML_`

The following is one way I can think of; I'm sure there are better ways to do it, this just shows the general idea:
```python
def get_key(prefix):
    # extract config key (also getting rid of yaml infix if needed)
for key, value in os.environ.items():
    config_key = None
    is_yaml = YAML_INFIX in key
    if key.startswith(CONFIG_PREFIX):
        config_key = get_key(CONFIG_PREFIX)
    elif key.startswith(OVERRIDE_PREFIX):
        config_key = get_key(OVERRIDE_PREFIX)
        if config_key in properties:
            continue  # we're done with this key! Maybe log a warning?
    if config_key:
        if is_yaml:
            # that's where the try/except goes (updating value)
        properties[config_key] = value
```
This *may be buggy* - I didn't run it. It's just a general example.

I suppose it is safe to assume that allowing yaml for both cases is reasonable (other reviewers might disagree though).
Also, would you mind adding a unit test? Just for this one function, it can go into `test/unit/util/test_properties.py`